src/snowflake/snowpark/mock/__init__.py

```diff
@@ -1,12 +1,32 @@
 #
 # Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
 #
+from json import JSONEncoder
+
 from ._functions import patch
 from ._snowflake_data_type import ColumnEmulator, ColumnType, TableEmulator
 
-CUSTOM_JSON_ENCODER = None
+
+class NumpyEncoder(JSONEncoder):
```

Comment:
may I ask what is this change for?

Reply:
This [chunk of code](https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/mock/_connection.py#L480-L482) was failing due to being unable to serialize numpy datatypes.

Reply:
that part of code is for use_xxx method on the session object, how would a UDF call trigger it?